src/Raven.Server/Documents/Handlers/BatchRequestParser.cs

```diff
@@ -104,10 +105,16 @@ private static void ReturnBuffer(CommandData[] cmds)
 
                     if (commandData.Type == CommandType.PATCH)
                     {
-                        commandData.PatchCommand = patcher.GetPatchDocumentCommand(commandData.Id, commandData.Etag, commandData.Patch, commandData.PatchIfMissing,
+                        commandData.PatchCommand = database.Patcher.GetPatchDocumentCommand(commandData.Id, commandData.Etag, commandData.Patch,
+                            commandData.PatchIfMissing,
                             skipPatchIfEtagMismatch: false, debugMode: false);
                     }
 
+                    if (commandData.Type == CommandType.PUT && commandData.Id?[commandData.Id.Length - 1] == '/')
```

Comment:
No, if `Id` is `null` then `[commandData.Id.Length - 1]` won't be evaluated. We'll have then the condition as follow: `null == '/'`